src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,213 @@
+package duke;
+
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDoTask;
+
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * The Parser class is responsible for parsing user input and executing
+ * the corresponding commands on the task list.
+ */
+public class Parser {
+
+    private static TaskList tasks;
+    private static Ui ui;
+
+    /**
+     * Parses the user input and executes the corresponding command.
+     *
+     * @param input The user's input command.
+     * @param tasks The TaskList instance to perform operations on.
+     * @param ui    The Ui instance for user interaction.
+     * @return The result of executing the command as a String.
+     * @throws DukeException If an error occurs during parsing or execution.
+     */
+    public static String parse(String input, TaskList tasks, Ui ui) throws DukeException {
+        Parser.ui = ui;
+        String[] parts = input.split(" ", 2);
+        String command = parts[0];
+        try {
+            switch (command) {
+            case "list":
+                return tasks.listTask();
+            case "delete":
+                int task = parseDeleteCommand(parts);
+                return tasks.deleteTask(task, ui);
+            case "todo":
+                ToDoTask toDoTask = parseTodoCommand(parts, false);
+                return tasks.addTask(toDoTask, ui);
+            case "deadline":
+                DeadlineTask deadlineTask = parseDeadline(parts[1], false);
+                return tasks.addTask(deadlineTask, ui);
+            case "event":
+                EventTask eventTask = parseEvent(parts[1], false);
+                return tasks.addTask(eventTask, ui);
+            case "mark":
+                int markTask = parseMarkCommand(parts) - 1;
+                if (markTask < 0 || markTask >= tasks.getTotalTasks()) {
+                    return ui.showNoTaskFound();
```

Comment:
I like the use of switch, but this if statement broke the "pattern" of calling and then returning a parsed command. Could this if statement perhaps be abstracted inside the parseMarkCommand and thrown as an error? Similar to how the delete command operates